src/visualization/src/VectorsVisualization.cpp

```diff
@@ -0,0 +1,224 @@
+{
+    // Delete existing node
+    if (m_vectors[vectorIndex].visualizationNode)
+    {
+        m_vectors[vectorIndex].visualizationNode->remove();
+        m_vectors[vectorIndex].visualizationNode = nullptr;
+    }
+
+    float arrowHeight = static_cast<float>(std::abs(m_vectors[vectorIndex].modulus) * m_heightScale);
+    float arrowWidth = static_cast<float>(m_radiusOffset +
+                                          m_radiusMultiplier * std::abs(m_vectors[vectorIndex].modulus) * m_heightScale);
+
+
+    auto arrowPosition = idyntree2irr_pos(m_vectors[vectorIndex].origin);
+
+    iDynTree::Direction vectorDirection = (m_vectors[vectorIndex].modulus < 0) ?
+                m_vectors[vectorIndex].direction.reverse() :
+                m_vectors[vectorIndex].direction;
+
+    iDynTree::Direction yDirection(0.0, 1.0, 0.0); //Arrows in irrlicht are pointing in the y direction by default
+
+    iDynTree::Direction rotationAxis;
+
+    iDynTree::toEigen(rotationAxis) = iDynTree::toEigen(yDirection).cross(iDynTree::toEigen(vectorDirection));
+    rotationAxis.Normalize();
+
+    double rotationAngle = std::acos(iDynTree::toEigen(vectorDirection).dot(iDynTree::toEigen(yDirection)));
+
+    iDynTree::Rotation arrowRotationMatrix = iDynTree::Rotation::RotAxis(rotationAxis, rotationAngle);
+
+    irr::core::vector3df arrowRotation = idyntree2irr_rot(arrowRotationMatrix);
+
+    irr::scene::ISceneNode * frameNode = m_smgr->addEmptySceneNode();
+
+    irr::scene::IMesh* arrowMesh = m_smgr->getGeometryCreator()->createArrowMesh(4, 8, arrowHeight, 0.9f * arrowHeight,
+                                                                                 arrowWidth, 2.0f * arrowWidth);
+
+    m_vectors[vectorIndex].visualizationNode = m_smgr->addMeshSceneNode(arrowMesh,frameNode);
+    m_vectors[vectorIndex].visualizationNode->setPosition(arrowPosition);
+    m_vectors[vectorIndex].visualizationNode->setRotation(arrowRotation);
+    irr::video::SMaterial arrowColor;
+    arrowColor.AmbientColor = idyntree2irrlicht(m_vectors[vectorIndex].color).toSColor();
+    arrowColor.DiffuseColor = arrowColor.AmbientColor;
+    m_vectors[vectorIndex].visualizationNode->getMaterial(0) = arrowColor;
+    m_vectors[vectorIndex].visualizationNode->getMaterial(1) = arrowColor;
+
+
+    arrowMesh->drop();
+    arrowMesh = nullptr;
+}
+
+void VectorsVisualization::drawAll()
+{
+    for (size_t i = 0; i < m_vectors.size(); ++i) {
+        drawVector(i);
+    }
+}
+
+VectorsVisualization::VectorsVisualization()
+    : m_smgr(nullptr)
+    , m_radiusOffset(0.01)
+    , m_radiusMultiplier(0.0)
+    , m_heightScale(1.0)
+{
+
+}
+
+void VectorsVisualization::init(irr::scene::ISceneManager *smgr)
+{
+    assert(smgr);
+    m_smgr = smgr;
+}
+
+void VectorsVisualization::close()
+{
+    for (auto& vector: m_vectors) {
+        if (vector.visualizationNode) {
+            vector.visualizationNode->removeAll();
+            //vector.visualizationNode->drop();
```

Comment:
This can be removed?

Reply:
Yes, it is a leftover of some trial for avoiding memory leaks.